Author: Esben Haabendal <esben@haabendal.dk>
Date:   Fri Nov 11 15:03:39 2016 +0100

    ifupdown: Add s6-init udhcpc service support
    
    This integrates with the udhcpc@* services generated by the net-dhcpc
    hook from netbase, enabling running udhcpc under s6-rc and thus under
    process monitoring.
    
    Signed-off-by: Esben Haabendal <esben@haabendal.dk>

Upstream-Status: Inappropriate [OE-lite/s6-init specific]

diff --git a/networking/ifupdown.c b/networking/ifupdown.c
index 7c45e89..296dc1e 100644
--- a/networking/ifupdown.c
+++ b/networking/ifupdown.c
@@ -521,6 +521,10 @@ struct dhcp_client_t {
 };
 
 static const struct dhcp_client_t ext_dhcp_clients[] = {
+	{ "service",
+		"service start udhcpc@%iface%",
+		"service stop udhcpc@%iface%",
+	},
 	{ "dhcpcd",
 		"dhcpcd[[ -h %hostname%]][[ -i %vendor%]][[ -I %client%]][[ -l %leasetime%]] %iface%",
 		"dhcpcd -k %iface%",
